crates/sui-protocol-config/src/lib.rs

```diff
@@ -675,6 +681,7 @@ impl ProtocolConfig {
 
                 max_tx_size: Some(64 * 1024),
                 max_tx_in_batch: Some(10),
+                max_gas_payment_objects: Some(128),
```

Comment:
I think making this low (like 32) to prevent doing free work of loading from DB is okay.